### PR TITLE
[CustomLoss] Update example

### LINKED ISSUE #1256: Custom Loss design and tests
- customLoss will be just like a regular custom layer
- For multiple losses, a constant value scalar for each loss needed
- requireLabel() in layer interface - then no  layer ahead of it in the graph

This is a subpart of #986

### PR DESCRIPTION
**Pull request by zhoonit**
- [CustomLoss] Update example

```
**Changes proposed in this PR:**
- Update example using LayerClient
- Add LayerClient example to AppTest

**Self evaluation:**
1. Build test: [X]Passed [ ]Failed [ ]Skipped
2. Run test: [X]Passed [ ]Failed [ ]Skipped

Signed-off-by: Jihoon Lee <jhoon.it.lee@samsung.com>

```

resolves #1256, #954

### CODE REVIEW

File: Applications/Custom/LayerClient/res/custom_layer_client.ini [hunk 1]
```diff
@@ -26,3 +25,6 @@ Type = fully_connected
 unit = 10
 Bias_initializer = zeros
 Activation = softmax
+
+[losslayer]
+Type = mae_loss # custom loss layer
```
Comment (kparichay) [inline Applications/Custom/LayerClient/res/custom_layer_client.ini:30]: i wonder if `loss = mae_loss` in `model` section also works?

Comment (kparichay) [reply Applications/Custom/LayerClient/res/custom_layer_client.ini:30]: If yes, add as a comment here? if no, then it needs debugging.

Comment (zhoonit) [reply Applications/Custom/LayerClient/res/custom_layer_client.ini:30]: The design here is that
1. Custom loss is not supported from model section(except the traditional basic ones)
2. We support loss as normal layers that is `requireLabel() == true`

To support multi loss, we aggregates loss from `requireLabel() == true`

So it is no bug but intended

Comment (kparichay) [reply Applications/Custom/LayerClient/res/custom_layer_client.ini:30]: > Custom loss is not supported from model section(except the traditional basic ones)

What about a custom loss from .so?

Comment (zhoonit) [reply Applications/Custom/LayerClient/res/custom_layer_client.ini:30]: Either from .so or directly from there source code which implements layer_devel.h

I'll update the comment inside ini file to cover the scenario you mentioned

**Review by kparichay — APPROVED** [review]
LGTM

File: Applications/Custom/LayerClient/res/custom_layer_client.ini [hunk 2]
```diff
@@ -26,3 +26,9 @@ Type = fully_connected
 unit = 10
 Bias_initializer = zeros
 Activation = softmax
+
+# Below is the custom loss layer,
+# This will directly replace loss in the Model section.
+# Thus, there should be no loss provided in the model section.
+[losslayer]
+Type = mae_loss
```
Comment (jijoongmoon) [inline Applications/Custom/LayerClient/res/custom_layer_client.ini:34]: I'm sure it it. But just check if there is type check between custom::mae_loss and mae_loss.

Comment (zhoonit) [reply Applications/Custom/LayerClient/res/custom_layer_client.ini:34]: Yes there is, if mae_loss exist, custom::mae_loss will fail to register. so the developer of mae_loss must have to choose different type name 👍

**Review by jijoongmoon — APPROVED** [review]
LGTM

**Review by lhs8928 — APPROVED** [review]
LGTM
